### PR TITLE
Fixed the params in ChoiceType methods

### PR DESCRIPTION
**Pull request by javiereguiluz**
This finishes #9173.

### CODE REVIEW

File: reference/forms/types/options/choice_label.rst.inc [hunk 1]
```diff
@@ -20,7 +20,7 @@ more control::
             'maybe' => null,
         ),
         'choices_as_values' => true,
-        'choice_label' => function ($value, $key, $index) {
+        'choice_label' => function ($category, $key, $value) {
```
Comment (vudaltsov) [inline reference/forms/types/options/choice_label.rst.inc]: Same here `($choiceValue, $key, $value)`.

File: reference/forms/types/options/group_by.rst.inc [hunk 2]
```diff
@@ -26,7 +26,7 @@ Take the following example::
             '1 month' => new \DateTime('+1 month'),
         ),
         'choices_as_values' => true,
-        'group_by' => function($value, $key, $index) {
+        'group_by' => function($vategory, $key, $value) {
```
Comment (vudaltsov) [inline reference/forms/types/options/group_by.rst.inc]: Same here `($choiceValue, $key, $value)`.

File: reference/forms/types/options/group_by.rst.inc [hunk 3]
```diff
@@ -26,7 +26,7 @@ Take the following example::
...
             if ($value <= new \DateTime('+3 days')) {
```
Comment (vudaltsov) [inline reference/forms/types/options/group_by.rst.inc]: Here the comparison should be against the first argument `$choiceValue`. I will check that for 2.8. But it's definitely true for 4.0. The first argument of the callback is the model data. The third `$value` is just a string from request.

**Review by vudaltsov — CHANGES_REQUESTED** [review]

File: reference/forms/types/options/choice_attr.rst.inc [hunk 4]
```diff
@@ -23,7 +23,7 @@ If an array, the keys of the ``choices`` array must be used as keys::
             'Maybe' => null,
         ),
         'choices_as_values' => true,
-        'choice_attr' => function($val, $key, $index) {
+        'choice_attr' => function($category, $key, $value) {
```
[the remark this answers is not in the document]
Comment (javiereguiluz) [reply reference/forms/types/options/choice_attr.rst.inc]: I'm going to do that change ... but then `'choice_attr' => function($choiceValue, $key, $value)` is going to look confusing. choiceValue and value? What's the difference?

File: reference/forms/types/choice.rst [hunk 5]
```diff
@@ -107,19 +107,19 @@ method::
...
         'choices_as_values' => true,
-        'choice_label' => function($category, $key, $index) {
+        'choice_label' => function($category, $key, $value) {
             /** @var Category $category */
             return strtoupper($category->getName());
         },
-        'choice_attr' => function($category, $key, $index) {
+        'choice_attr' => function($category, $key, $value) {
             return ['class' => 'category_'.strtolower($category->getName())];
         },
```
Comment (vudaltsov) [inline reference/forms/types/choice.rst]: empty line

File: reference/forms/types/options/choice_label.rst.inc [hunk 6]
```diff
@@ -20,7 +20,7 @@ more control::
             'maybe' => null,
         ),
         'choices_as_values' => true,
-        'choice_label' => function ($value, $key, $index) {
+        'choice_label' => function ($choiceValue, $key, $value) {
             if ($value == true) {
```
Comment (vudaltsov) [inline reference/forms/types/options/choice_label.rst.inc]: I think that is makes more sense to check `true === $choiceValue` here.

**Review by vudaltsov — CHANGES_REQUESTED** [review]
Sorry, found some other things that could be fixed in this PR. And some other improvements I will do in a new one.

**Comment by javiereguiluz** [comment]
@vudaltsov I thought you were going to create a new PR instead of this one ... so I was waiting :)  I'm going to finish and merge this so you can create a new PR. Thanks!

**Comment by javiereguiluz** [comment]
This is now merged! @vudaltsov when you create your PR, please make any change needed to create variables easy to understand. E.g. `$choiceValue` -> is the value chosen by the user or the value shown in the choice list? Better use `$selectedValue` or `$selectedIndex`, etc. Thanks!

**Comment by vudaltsov** [comment]
I think that `$choiceValue` is understandable enough because that's the value from the `choices` option. We can rename it only when we know what it is in terms of model data (like `$category`).

**Comment by yceruto** [comment]
@vudaltsov what about just `$choice`? that's how it is called internally for all cases:
https://github.com/symfony/symfony/blob/3a2eb0d9514f0b322ab31a128aa48b9f89e28d50/src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php#L121

https://github.com/symfony/symfony/blob/3a2eb0d9514f0b322ab31a128aa48b9f89e28d50/src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php#L130

https://github.com/symfony/symfony/blob/3a2eb0d9514f0b322ab31a128aa48b9f89e28d50/src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php#L140

https://github.com/symfony/symfony/blob/3a2eb0d9514f0b322ab31a128aa48b9f89e28d50/src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php#L144

https://github.com/symfony/symfony/blob/3a2eb0d9514f0b322ab31a128aa48b9f89e28d50/src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php#L203

**Comment by vudaltsov** [comment]
@yceruto , okay! I will do that in my PR.

**Comment by javiereguiluz** [comment]
I still think it's not understandable. I can't understand anything here --> `function ($choice, $index, $value)`

Please, let's pick variable names that the reader cannot mistake. For example, which of the following names best describe the previous arguments?

```php
function ($selectedChoice, $choiceIndex, $choiceValue)
function ($availableChoice, $selectedIndex, $selectedValue)
function ($availableChoice, $choiceIndex, $choiceValue)
etc.
```

**Comment by vudaltsov** [comment]
The problem with `$choiceValue` is that we have an option `choice_value` which is about `$viewValue` and that might be misleading. While `data` indicates that this is more of a model thing, the actual choice data.

**Pull request closed without merging**